### PR TITLE
[Improve] support delete flink home

### LINKED ISSUE #2542: [Feature] Support delete Flink home.
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-streampark/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

[Feature] Support delete Flink version.

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zhoulii**
## What changes were proposed in this pull request

Issue Number: close #2542 

## Brief change log

after

## Verifying this change

- *Manually verified the change by testing locally.* 

## Does this pull request potentially affect one of the following parts
 - Dependencies (does it add or upgrade a dependency): (yes / **no**)

### CODE REVIEW

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/FlinkEnvServiceImpl.java [hunk 1]
```diff
@@ -39,6 +42,8 @@
 public class FlinkEnvServiceImpl extends ServiceImpl<FlinkEnvMapper, FlinkEnv>
     implements FlinkEnvService {
 
+  @Autowired private FlinkClusterService flinkClusterService;
+  @Autowired private ApplicationService applicationService;
```
Comment (RocMarshal) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/FlinkEnvServiceImpl.java:46]:
```suggestion
  @Autowired private ApplicationService applicationService;
  
```

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/ApplicationServiceImpl.java [hunk 2]
```diff
@@ -533,6 +533,13 @@ public boolean existsJobByClusterId(Long clusterId) {
     return baseMapper.existsJobByClusterId(clusterId);
   }
 
+  @Override
+  public boolean existsJobByFlinkEnvId(Long flinkEnvId) {
+    LambdaQueryWrapper<Application> lambdaQueryWrapper =
+        new LambdaQueryWrapper<Application>().eq(Application::getVersionId, flinkEnvId);
+    return this.count(lambdaQueryWrapper) > 0;
```
Comment (RocMarshal) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/ApplicationServiceImpl.java]: May here use `getBaseMapper() .exists(....` ?

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/FlinkClusterServiceImpl.java [hunk 3]
```diff
@@ -323,6 +323,13 @@ public Boolean existsByClusterName(String clusterName, Long id) {
     return this.baseMapper.existsByClusterName(clusterName, id);
   }
 
+  @Override
+  public Boolean existsByFlinkEnvId(Long flinkEnvId) {
+    LambdaQueryWrapper<FlinkCluster> lambdaQueryWrapper =
+        new LambdaQueryWrapper<FlinkCluster>().eq(FlinkCluster::getVersionId, flinkEnvId);
+    return this.count(lambdaQueryWrapper) > 0;
```
Comment (RocMarshal) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/FlinkClusterServiceImpl.java]: same as mentioned above.

File: streampark-console/streampark-console-webapp/src/locales/lang/zh-CN/setting/flinkHome.ts [hunk 4]
```diff
@@ -18,6 +18,8 @@ export default {
   title: 'Flink Home',
   conf: 'Flink 配置',
   sync: '配置同步',
+  edit: '编辑 Flink Home',
+  delete: '确定要删除此 Flink home ?',
```
Comment (RocMarshal) [inline streampark-console/streampark-console-webapp/src/locales/lang/zh-CN/setting/flinkHome.ts:22]: I checked the  header of the edit tab window is :
image

**Review by RocMarshal — COMMENTED** [review]
@zhoulii Thanks for your contribution & quick-drive
I left a few of comments.  PTAL in your free time.

May the items  from my limit read as follows get your @wolfboys @zhoulii confirmation ?

1. If there are running jobs or sessions using the target flinkhome, it would be better not to support changing Flinkhome to avoid abnormal risks when some savepoints are triggered.
2. The `delete`,`update`,`create` api have  no `ApiAccess` or `XXXPermission` annotations . Is here any risk ?

I'd appreciated with your confirmation.
thanks~

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/FlinkEnvServiceImpl.java (hunk 1 above)
Comment (zhoulii) [reply streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/FlinkEnvServiceImpl.java:46]: done

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/ApplicationServiceImpl.java (hunk 2 above)
Comment (zhoulii) [reply streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/ApplicationServiceImpl.java]: done

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/FlinkClusterServiceImpl.java (hunk 3 above)
Comment (zhoulii) [reply streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/FlinkClusterServiceImpl.java]: done

File: streampark-console/streampark-console-webapp/src/locales/lang/zh-CN/setting/flinkHome.ts (hunk 4 above)
Comment (RocMarshal) [reply streampark-console/streampark-console-webapp/src/locales/lang/zh-CN/setting/flinkHome.ts:22]: `添加`-> `编辑`  / `Add `->`Edit`

Comment (zhoulii) [reply streampark-console/streampark-console-webapp/src/locales/lang/zh-CN/setting/flinkHome.ts:22]: This is due to Adding and Editing Flink Home use the same [modal#L169](https://github.com/apache/incubator-streampark/blob/dev/streampark-console/streampark-console-webapp/src/views/setting/FlinkHome/components/Modal.vue#L169), and the title is fixed. I have limited understanding of frontend, what I can do is change the title to `Add/Edit`. but maybe it's not a elegant way.

**Comment by zhoulii** [comment]
> @zhoulii Thanks for your contribution & quick-drive I left a few of comments. PTAL in your free time.
> 
> May the items from my limit read as follows get your @wolfboys @zhoulii confirmation ?
> 
> 1. If there are running jobs or sessions using the target flinkhome, it would be better not to support changing Flinkhome to avoid abnormal risks when some savepoints are triggered.
> 2. The `delete`,`update`,`create` api have  no `ApiAccess` or `XXXPermission` annotations . Is here any risk ?
> 
> I'd appreciated with your confirmation. thanks~

@RocMarshal Thanks for the review. I agree with you, maybe we can hanle these in another pr, and let this one focus on delete operation. what do you think?

**Comment by RocMarshal** [comment]
> I agree with you, maybe we can hanle these in another pr, and let this one focus on delete operation. what do you think?
Thanks for the reply~
SGTM If we can do it before the next release.
looking forward to it~
CC @wolfboys .

**Comment by wolfboys** [comment]
> > I agree with you, maybe we can hanle these in another pr, and let this one focus on delete operation. what do you think?
> > Thanks for the reply~
> > SGTM If we can do it before the next release.
> > looking forward to it~
> > CC @wolfboys .

I agree with you,  We can do it in next pr.

File: streampark-console/streampark-console-webapp/src/locales/lang/zh-CN/setting/flinkHome.ts (hunk 4 above)
Comment (RocMarshal) [reply streampark-console/streampark-console-webapp/src/locales/lang/zh-CN/setting/flinkHome.ts:22]: @zhoulii Take it easy, please ~ 😁
Hi, @wangsizhu0504 Could you help take a look about this ?
Thanks a lot~

File: streampark-console/streampark-console-webapp/src/views/setting/FlinkHome/index.vue [hunk 5]
```diff
@@ -61,6 +79,13 @@
     openEnvDrawer(true, res);
   }
 
+  /* delete flink home */
+  async function handleDelete(item: FlinkEnv) {
+    await fetchFlinkEnvRemove(item.id);
+    await getFlinkSetting();
+    createMessage.success('The current flink home is removed');
+  }
```
Comment (wolfboys) [inline streampark-console/streampark-console-webapp/src/views/setting/FlinkHome/index.vue:89]:
```
 async function handleDelete(item: FlinkEnv) {
    const resp = await fetchFlinkEnvRemove(item.id);
    if (resp.data.code == 200) {
      await getFlinkSetting();
      createMessage.success('The current flink home is removed');
    }
  }

```

**Review by wolfboys — APPROVED** [review]
LGTM

**Pull request merged**